tests/test.py

```diff
@@ -0,0 +1,184 @@
+
```

Comment:
Please add a header to the file and rename to `test_while.py`. You will also need to move it to `tests/qasm3/` folder